Feature/#34/combine config parameters

felixble opened the pull request:
Related issues: #34 and #9

dobecker1 committed:
npm script for running e2e tests

felixble committed:
Define `.env.sample` file with sample configuration / apply for shell scripts

* used by vagrant: after-up.sh and bootstrap.sh (both tested locally)
* .env file will be set up once after the vm starts (copies sample file)
* add `.env` to `.gitignore`

felixble committed:
Use `.env` in our web application and get rid of `local.php`

Keep in mind: `composer install` require to install new dependency

felixble committed:
Merge remote-tracking branch 'origin/feature/#9/setup-package.json' into feature/#34/combine-config-parameters

# Conflicts:
#	test/e2e/config.js
#	test/e2e/run.sh

felixble committed:
Use `.env` in our js scripts (node utils / e2e tests)

Uses npm module `env-cmd` to load the config as environment variables before executing the scripts / tests

felixble committed:
Default env configuration for travis

felixble committed:
Run unit tests on travis, too

felixble committed:
Added missing npm script to run our unit tests

jasteuer commented on .travis.yml:
```diff
@@ -28,10 +28,10 @@ before_install:
 script:
-  - testcafe chrome,firefox test/**/*.test.js --app "php -S 0.0.0.0:8888 -t ./"
\ No newline at end of file
+  - testcafe chrome,firefox test/**/*.test.js --app "php -S 0.0.0.0:8888 -t ./"
```
is it possible to use one of the npm scripts (`npm run test:e2e:XXX`) here?

jasteuer commented on test/e2e/run.sh:
```diff
@@ -2,7 +2,7 @@
+## start the php server
 # php -S localhost:8888 -t ../../ &
 # SERVER_PID=$!
 
```
I think this file is no longer necessary:
- developer starts server with vagrant and use npm scripts to run tests
- travis starts server and run tests according to its configuration

felixble replied:
Yep, you are right. I will remove it

jasteuer commented on .travis.yml:
```diff
@@ -28,10 +28,10 @@ before_install:
 script:
-  - testcafe chrome,firefox test/**/*.test.js --app "php -S 0.0.0.0:8888 -t ./"
\ No newline at end of file
+  - testcafe chrome,firefox test/**/*.test.js --app "php -S 0.0.0.0:8888 -t ./"
```
is it possible to start the server and then use one of the npm scripts (`npm run test:e2e:XXX`) here?

felixble replied:
Good idea. Thanks!

jasteuer reviewed (changes requested):
Looks good! Only small things I requested to change

felixble committed:
Removed `test/e2e/run.sh` / Start e2e tests on travis via npm script

The shell script is not necessary anymore since developer starts server with vagrant and use npm scripts to run tests and travis starts server and run tests according to its configuration

felixble committed:
Travis ci install npm dependencies / Move npm dependencies to dev-dependencies

We use node (and all npm dependencies) only for development reasons so they should be listed as dev-dependencies.

felixble commented:
@jasteuer Thanks for your review. I have implemented the requested changes. Please do not forget to append "Closes #34 and closes #9" to the commit message when merging this PR since both tickets are done with this PR.
